chain/chain/src/store.rs

```diff
@@ -1745,6 +1772,27 @@ impl<'a> ChainStoreUpdate<'a> {
             .insert((height, shard_id), chunk_hash);
     }
 
+    pub fn inc_block_refcount(&mut self, block_hash: &CryptoHash) -> Result<(), Error> {
+        let refcount = match self.get_block_refcount(block_hash) {
+            Ok(refcount) => refcount.clone(),
+            Err(e) => match e.kind() {
+                ErrorKind::DBNotFoundErr(_) => 0,
+                _ => return Err(e),
+            },
+        };
+        self.chain_store_cache_update.block_refcounts.insert(*block_hash, refcount + 1);
+        Ok(())
+    }
+
+    // TODO will be in use in #2084
+    #[allow(unused)]
+    pub fn dec_block_refcount(&mut self, block_hash: &CryptoHash) -> Result<(), Error> {
+        let refcount = self.get_block_refcount(block_hash)?.clone();
+        assert!(refcount > 0);
```

Comment:
I think this assert is unnecessary. It is better to return an error so that the node doesn't crash when this happens.

Reply:
I've changed it as you suggested, according to our slack discussion.